client/src/components/timetable/ExportPage.jsx

```diff
@@ -0,0 +1,61 @@
+import { Component } from "react";
+import React from "react";
+
+import PreviewTT from "./PreviewTT.jsx";
+import MidsemSched from "./MidsemSched.jsx";
+import CompreSched from "./CompreSched.jsx";
+
+import html2canvas from "html2canvas";
+import jsPDF from "jspdf";
+
+class ExportPage extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  componentDidMount() {
+    const timetableComp = document.getElementById("timetable-pdf");
+    const pageHeight = timetableComp.offsetHeight;
+    const pageWidth = timetableComp.offsetWidth;
+    const midsemComp = document.getElementById("midsem-pdf");
+    const secondPageOffset = midsemComp.offsetHeight;
+    const compreComp = document.getElementById("compre-pdf");
+    const pdf = new jsPDF('landscape', 'px',
+      [pageHeight, pageWidth]
+    );
```

Comment:
Problem seems to be with this `pageHeight` and `pageWidth`. Don't exactly know what it is. Are you sure it takes the preview's dimensions and not the normal displays, because the cutoff region seems very close to that.

Reply:
I use the offsets of the preview element. Moreover, if I save as image, instead of making a pdf, there is no whitespace there